### ISSUE TITLE
Do you have to explicitly end a stream?

### ISSUE THREAD

**Issue by pzolla** [paldepind/flyd#110]
If I have a stream:

var number = flyd.stream(5);

And I decide I need to reassign the variable 'number', can I simply do so?  Or should you end the stream and then reassign?

**Comment by c-dante** [comment]
So, stream updates are atomic in that there exists a dependency graph when you create dependent streams.

```js
a = stream
b = stream
c = combine (a, b)
d = map c
/*
A   B
 \ /
  C
  |
  D
*/
```

In javascript, garbage collection will only clean objects that have no other references. So in this example, if you later do:

```js
c = something else
```

Then `D` will still be calculated, but you won't be able to get a reference to `C` any more.

If you do:
```js
c.end(true)
c =  something else
```

Then `D` will be ended and no longer receive values, and you won't have a reference to `C` any more.

It depends what goal you have.

What's more important is you shouldn't be declaring a variable and then later changing its type -- it's bad form and you end up with messy, hard to follow code. "Is `number` a stream? Or an array? What is it at this point in code?" But that's more how you use JS as a language, not how you use flyd.

**Comment by pzolla** [comment]
Thanks for your response.  My use case was a bit vague.  I am not changing type, so much as resetting everything to a new empty stream.  That said, you did provide the insight I needed to understand the impact of 'end' and how it effects garbage collection.
